sql-plugin/src/main/spark400/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "400"}
+spark-rapids-shim-json-lines ***/
+package org.apache.spark.sql.rapids.shims
+
+import org.apache.spark.sql.AnalysisException
+
+object RapidsErrorUtils extends RapidsErrorUtilsFor340Plus {
+  override def dynamicPartitionParentError: Throwable = {
+    /** This exception doesn't have a helper method so the errorClass has to be hardcoded */
+    throw new AnalysisException(
+      errorClass = "_LEGACY_ERROR_TEMP_3079",
```

Comment:
I'm worried about this changing out from underneath us and then the error doesn't work at all.  The name itself implies this is temporary and will be changing, yet the compiler won't catch when that occurs.  It might be better to throw an error message directly here for now rather than rely on this tenuous connection to Spark's error message.

Reply:
I have replaced this with the current way of throwing `ErrorMsg.PARTITION_DYN_STA_ORDER.getMsg`.